tools/vscode/generate_debug_config.py

```diff
@@ -89,20 +89,38 @@ def lldb_config(target, binary, workspace, execroot, arguments):
     }
 
 
-def add_to_launch_json(target, binary, workspace, execroot, arguments, debugger_type):
-    launch = get_launch_json(workspace)
+def get_old_config(config_name, configurations):
```

Comment:
i dont think we need any of this - why not just replace

https://github.com/envoyproxy/envoy/blob/cda376394e1b73dc337e0f40c3b5560d2c338321/tools/vscode/generate_debug_config.py#L100-L107

with something like

```python

 configurations = launch.get("configurations", []) 
 always_overwritten_fields = ["x", "y", "z"]
 for config in configurations: 
     if config.get("name", None) == new_config["name"]: 
         if overwrite:
             config.clear() 
             config.update(new_config) 
         else:
             for k in always_overwritten_fields:
                 config[k] = new_config[k]
         break 
 else: 
     configurations.append(new_config) 

```

Reply:
agree, maybe `always_preserve_fields` is better?
```python
 configurations = launch.get("configurations", []) 
 always_preserve_fields = ["args"]
 for config in configurations: 
     if config.get("name", None) == new_config["name"]: 
         if overwrite:
             config.clear() 
             config.update(new_config) 
         else:
             for k not in always_preserve_fields:
                 old_config[k] = new_config[k]
             print(f"old config exists, fields {always_preserve_fields} will be preserved, use --overwirte to recreate config")
         break 
 else: 
     configurations.append(new_config) 
```

Reply:
> maybe always_preserve_fields is better?

wfm

Reply:
hmm - actually not `preserve` - its not clear what is being preserved ie old/new config

how about `always_update_fields` ?

Reply:
hh, for old config is preserve, for new config is update.

Reply:
tbh i think my preference is towards `always_overwrite_fields` as it matches to `--overwrite`